### PR TITLE
Authentication feature

### PR DESCRIPTION
**Pull request by Hugovs2000**
Implemented Authentication using firebase and tested it with simple landing, log in and sign up components.

### CODE REVIEW

**Review by yanevdw — APPROVED** [review]
Auth looks good. Well done!

File: src/app/app.routes.ts [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+    component: LandingComponent,
+  },
+  {
+    path: 'log-in',
+    component: LoginComponent,
+  },
+  {
+    path: 'sign-up',
+    component: SignupComponent,
+  },
```
Comment (Musa-Mabasa) [inline src/app/app.routes.ts:31]: Don't forget to add an error page and a redirect to the landing component.

File: src/app/components/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  });
+
+  hide = true;
+
+  onSubmit() {
+    if (this.loginForm.value.email && this.loginForm.value.password)
+      this.authService
+        .login(this.loginForm.value.email, this.loginForm.value.password)
+        .then(() => this.router.navigate(['']))
+        .catch(() =>
```
Comment (Musa-Mabasa) [inline src/app/components/login/login.component.ts:54]: Quick suggestion, you could use a notification to alert users of the error as it is important for them to know what went wrong so they could rectify the mistake if it was from their side. Same goes for the other auth functions.

File: src/app/components/signup/signup.component.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      confirmPassword: new FormControl('', [Validators.required]),
+    },
+    this.validation.passwordMatch('password', 'confirmPassword')
+  );
+
+  onSubmit() {
+    if (this.signUpForm.value.email && this.signUpForm.value.password)
+      this.authService
+        .signup(this.signUpForm.value.email, this.signUpForm.value.password)
+        .then(() => this.router.navigate(['log-in']));
```
Comment (Musa-Mabasa) [inline src/app/components/signup/signup.component.ts]: Same thing here with the notification alerts incase of an error

**Review by Musa-Mabasa — COMMENTED** [review]
If notification alerts for auth are intended for another branch then let me know so I can approve, if not, then changes need to be made.

**Review by Musa-Mabasa — APPROVED** [review]
Everything looks good 👍

**Pull request merged**